dashboard/cron.py

```diff
@@ -656,6 +648,9 @@ def do(self):
         if courses_added_during_cron:
             logger.warning(f'During the run, users added {len(courses_added_during_cron)} course(s): {courses_added_during_cron}')
             logger.warning(f'No data was pulled for these courses.')
+            # Set all of the courses to have been updated now (this is the same set update_course runs on)
+
+        Course.objects.get_supported_courses().update_all_last_cron_run()
```

Comment:
I don't think this is what you want. I think you want `Course.objects.filter(self.valid_locked_course_ids).update_all_last_cron_run()`, or something similar. You don't want to update courses added during the cron.

Aside from the course set you are updated, I'm a little confused by the chaining of `get_supported_courses().update_all_last_cron_run()`. Doesn't `get_supported_courses` return a list of `int`s and `update_all_last_cron_run` act on a `QuerySet`?

Sorry if I'm missing something here. I'm not that familiar with defining custom `QuerySet` objects.

Reply:
Hm, looks like it's still a `QuerySet`, it just evaluates to a list of `int`s instead of a list of `Course` objects, I guess.

Reply:
I'm still concerned about the use of `Course.objects.get_supported_courses` here. If a course is added during the cron, you're going to give it a `last_cron_run` value, even though it wasn't updated, right?

Reply:
Yeah, I think I was back and forth between putting it into this block but just wanted something working. I'll look at this and change it to your suggestion.

Reply:
Yeah, it returns a QuerySet which seems convenient for chaining other calls. 

I could write it like you have it though if it was in the if statement.

Reply:
I guess it doesn't need to be, but if there aren't any then it won't do anything. So I guess same thing.

Reply:
Yeah, I see what you mean, it should just update the valid_locked ones, so it's fine where it is.